envctl/env_scripts/java/kubernetes.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e # exit on any failure
+set -o pipefail # any step in pipe caused failure
+set -u # undefined variables cause exit
+
+SERVICE_NAME="log-java-gke-$(echo $ENVCTL_ID | head -c 8)"
+ZONE=us-central1-a
+
+destroy() {
+  set +e
+  # delete pubsub resources
+  gcloud pubsub topics delete $SERVICE_NAME -q 2> /dev/null
+  gcloud pubsub subscriptions delete $SERVICE_NAME-subscriber -q 2> /dev/null
+  # delete container images
+  export GCR_PATH=gcr.io/$PROJECT_ID/logging:$SERVICE_NAME
+  gcloud container images delete $GCR_PATH -q --force-delete-tags 2> /dev/null
+  # delete cluster
+  gcloud container clusters delete --zone $ZONE $SERVICE_NAME -q
+  set -e
+}
+
+verify() {
+  set +e
+  gcloud pubsub subscriptions describe $SERVICE_NAME-subscriber 2> /dev/null
+  if [[ $? != 0 ]]; then
+     echo "FALSE"
+     exit 1
+  fi
+  gcloud container clusters describe --zone $ZONE $SERVICE_NAME > /dev/null 2> /dev/null
+  if [[ $? == 0 ]]; then
+     echo "TRUE"
+     exit 0
+   else
+     echo "FALSE"
+     exit 1
+  fi
+  set -e
+}
+
+attach_or_create_gke_cluster(){
```

Comment:
I wonder if it wouldn't be better to have a cluster pre-created for tests and only create a dedicated namespace and deployments in it for each test run?

Reply:
That would make the test faster, but the downside is it would be another thing we'd need to manually maintain, and the ongoing resources would be much more expensive when we only really need it live for 5 mins a day

The environment tests run in parallel, and the GKE test takes around the same time as App Engine Standard (~10 mins), so I don't think speeding it up should be too much of a priority

Reply:
In cloud-code-samples we had pre-created gke cluster and each test run created a dedicated namespace on GKE and all the test objects belonged go that. Then there was a periodic task that was deleting all the namespaces that weren't deleted automatically by the test runs.
For now we can keep this approach in order to unblock the PR. However going forward we might reconsider - depending on how stable is provisioning before the actual tests.